packages/core/src/application.ts

```diff
@@ -41,9 +55,11 @@ export class Application extends Context {
    */
   controller(controllerCtor: ControllerClass, name?: string): Binding {
     name = name || controllerCtor.name;
-    return this.bind(`controllers.${name}`)
+    const key = `controllers.${name}`;
+    debug('Adding controller %s', name);
+    return this.bind(key)
       .toClass(controllerCtor)
-      .tag('controller');
+      .tag(CoreBindings.CONTROLLER_TAG);
```

Comment:
It would be great if you could extract the part reworking existing binding tags from strings to constants and send a standalone pull request.